types/wordpress__data/wordpress__data-tests.ts

```diff
@@ -0,0 +1,25 @@
+import * as data from "@wordpress/data";
+
+data.select("core/block-editor").isTyping<boolean>();
```

Comment:
The `<boolean>` here amounts to assertion that the result will be a `boolean`.  Since that type isn't either inferred or checked, it might be clearer to explicitly assert that it's `boolean`.

Reply:
Can't do that because there's no way to know ahead of time of all the possibilities that `select` may return.

See [here](https://github.com/dsifford/academic-bloggers-toolkit/blob/master/types/augments.d.ts) for an example of how a user might augment the definitions to be more type-safe.